test/cmdlineTests/standard_ast_export_after_fatal_analysis_error/input.json

```diff
@@ -0,0 +1,15 @@
+{
+	"language": "Solidity",
+	"sources":
+	{
+		"A":
+		{
+			"content": "// SPDX-License-Identifier: GPL-3.0\npragma solidity >=0.0;\nstruct S { S s; } contract C { event E(S); }"
+		}
+	},
+	"settings":
+	{
+		"parserErrorRecovery": true,
```

Comment:
This case, with error recovery enabled, actually ICEs without the change:

The AST export for the event will try to include the event signature, in case analysis was successful.
However the recursive struct is a fatal error, which, incorrectly, set the "analysis performed" state, tricking the AST export into thinking type information is available, resulting in an ICE on the event export.

Reply:
Should we even be allowing export in case of failed analysis?

Reply:
What we should do is drop the broken error recovery mode in general - it's full of all sorts of issues like this :-).